name: Conformance Tests

on:
  push:
    branches:
      - main
  workflow_dispatch:
  pull_request:

jobs:
  conformance:
    runs-on: ubuntu-latest
    steps:
      - uses: actions/checkout@b4ffde65f46336ab88eb53be808477a3936bae11 # v4.1.1

      - uses: actions/setup-python@0a5c61591373683505ea898e09a3ea4f39ef2b9c # v5.0.0
        with:
          python-version: "3.x"
          cache: "pip"
          cache-dependency-path: pyproject.toml

      - name: install sigstore-python
        run: python -m pip install .

      - uses: sigstore/sigstore-conformance@c8d17eb7ee884cf86b93a3a3f471648fb0a83819 # v0.0.9
        with:
          entrypoint: ${{ github.workspace }}/test/integration/sigstore-python-conformance
          xfail: "test_verify_with_trust_root" # see issue 821
